samples/Network/hdinsight-require-subscription-vnet/azurepolicy.json

```diff
@@ -0,0 +1,56 @@
+{
+    "mode": "all",
+    "policyRule": {
+      "if": {
+        "anyOf": [
+          {
+            "allOf": [
+              {
+                "field": "type",
+                "equals": "Microsoft.HDInsight/clusters"
+              },
```

Comment:
This type check is repeated.

Could be simplified by having the "allOf" on the outside, with the "type" condition first, then "anyOf" inside the second condition

Reply:
`
     "allOf": [
        {
          "field": "type",
          "equals": "Microsoft.HDInsight/clusters"
        },
        {
          "anyOf": [
            {
              "not": {
                "field": "Microsoft.HDInsight/clusters/computeProfile.roles[*].virtualNetworkProfile.subnet",
                "notEquals": ""
              }
            },
            {
              "field": "Microsoft.HDInsight/clusters/computeProfile.roles[*].virtualNetworkProfile.subnet",
              "exists": "false"
            }
          ]
        }
      ]`

---
In reply to: [268248473](https://github.com/Azure/azure-policy/pull/340#discussion_r268248473) [](ancestors = 268248473)